onnxruntime/contrib_ops/cpu/bert/multihead_attention.cc

```diff
@@ -293,20 +293,6 @@ Status MultiHeadAttention<T>::Compute(OpKernelContext* context) const {
 
   AllocatorPtr allocator;
 
-  if (qkv_bias == nullptr) {
-    // We assume query, key/past_key, and value/past_value are already in the correct shape
-
-    // Check that key/value or past_key/past_value is valid
-    ORT_ENFORCE((key != nullptr && value != nullptr) || (past_key != nullptr && past_value != nullptr));
-    return ApplyAttention(query->Data<T>(),
```

Comment:
Removing this code will cause Transpose to be invoked when there's no qkv bias, which changes the shape of Q/K/V.  Have you accounted for both scenarios to maintain backwards compatibility: 1) when Q/K/V is already in the correct input shape and 2) when Q/K/V is not in the correct input shape?

Reply:
I think the latter portion of the code is ensuring the layout. This portion is using incorrect layout qkv for ApplyAttention and is causing testing failures.

Reply:
@kunal-vaishnavi, could you verify that the change is good for Whisper model?